src/jaegertracing/TracerFactory.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "TracerFactory.h"
+
+#include "jaegertracing/Tracer.h"
+
+namespace jaegertracing {
+
+opentracing::expected<std::shared_ptr<opentracing::Tracer>>
+TracerFactory::MakeTracer(const char* configuration,
+                          std::string& errorMessage) const noexcept try {
+    YAML::Node yaml;
```

Comment:
I think I have the whole YAML dependency wrapped in a macro at the moment. If so, can you add this to wrap the existing code.

```
#ifdef JAEGERTRACING_USE_YAML_CPP
// existing code
#else
// return dummy value or construct config by hand
#endif  // JAEGERTRACING_USE_YAML_CPP
```

Reply:
I set up dynamic loading to fail if YAML support isn't compiled in and added some test coverage for it as well.